refactoring get_cpu_pinning() (#22489)
openvinotoolkit/openvino: docs/articles_en/openvino_workflow/running_inference_with_openvino/Device_Plugins/CPU.rst

Reviewer: I see the following statement in this doc: 
```ov::hint::enable_cpu_pinning only support Linux in current release.```
Is it still actual?

Author: Updated

user can check the :doc:`optimization guide <openvino_docs_deployment_optimization_guide_tput_advanced>` for details on multi-stream execution

.. note::

>    ``ov::hint::enable_cpu_pinning`` is not supported on Windows dual-socket XEON platform in current release.

Reviewer: ```suggestion
   ``ov::hint::enable_cpu_pinning`` is not supported on multi-socket platforms with Windows OS.
```

Author: updated
